arrow-core-data/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -363,7 +365,11 @@ import arrow.typeclasses.Show
  * Contents partially adapted from [Scala Exercises Option Tutorial](https://www.scala-exercises.org/std_lib/options)
  * Originally based on the Scala Koans.
  */
-
+@Deprecated(
+  "Option will be deleted soon as it promotes the wrong message of using a slower and memory unfriendly " +
+    "abstraction when the lang provides a better one.",
+  ReplaceWith("Either<Unit, A>")
```

Comment:
I think the memory friendly replacement is using Kotlin's nullable types. The suggestion for `Either<Unit, A>` was for places where that is not possible.

Reply:
yes, we need to favor nullable types until a use case is proven. Option is not kotlin idiomatic and confuses people, when it's used for internals it can always be replaced by Either<Unit, A>

Reply:
thanks for the tip @LordRaydenMK! changed now